app/controllers/posts_controller.rb

```diff
@@ -1,7 +1,7 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
-    @posts = Post.where(user_id: params[:user_id])
+    @posts = Post.includes(:user).where(user_id: params[:user_id])
```

Comment:
- [ ] Please solve N+1 problem when fetching all posts and their comments for a user. You could do something like this instead 👇 
```ruby
@posts = @user.posts.includes(:comments)
```